asdf/block.py

```diff
@@ -823,16 +824,27 @@ def array_storage(self):
         return self._array_storage
 
     @property
-    def compression(self):
-        return self._compression
+    def input_compression(self):
+        """
+        The compression codec used to read the block.
+        """
+        return self._input_compression
 
-    @compression.setter
-    def compression(self, compression):
-        self._compression = mcompression.validate(compression)
+    @input_compression.setter
+    def input_compression(self, compression):
+        self._input_compression = mcompression.validate(compression)
 
     @property
-    def is_compressed(self):
-        return self._compression is not None
+    def output_compression(self):
+        """
+        The compression codec used to write the block.
+        :return:
+        """
+        return self._output_compression
```

Comment:
In order to account for the case where a file with compression is read, but we do not want to have to explicitly say `all_array_compression` on output, I think this function should look something like this:
```python
def output_compression(self):
    if self._output_compression is None and self._input_compression is not None:
        return self._input_compression
    return self._output_compression
```
I haven't looked at this as closely as you have, but does this make sense?